core/src/main/java/apoc/nodes/Grouping.java

```diff
@@ -426,6 +462,145 @@ private void aggregate(Entity pc, Map<String, List<String>> aggregations, Map<St
         });
     }
 
+    private Object getMin(String key, Entity pc, Object value) {
+        Object prop = pc.getProperty(key);
+
+        if (prop == null) {
+            return value;
+        }
+
+        if (isComparableTypes(prop, value)) {
+            return compareValues(prop, value) ? prop : value;
+        }
+
+        return returnMinOfDifferentValues(prop, value);
+    }
+
+    private Object getMax(String key, Entity pc, Object value) {
+        Object prop = pc.getProperty(key);
+
+        if (prop == null) {
+            return value;
+        }
+
+        if (isComparableTypes(prop, value)) {
+            return compareValues(prop, value) ? value : prop;
+        }
+
+        return returnMaxOfDifferentValues(prop, value);
+    }
+
+    private boolean isComparableTypes(Object prop, Object value) {
+        return (prop instanceof ZonedDateTime && value instanceof ZonedDateTime)
+                || (prop instanceof LocalDateTime && value instanceof LocalDateTime)
+                || (prop instanceof LocalDate && value instanceof LocalDate)
+                || (prop instanceof OffsetTime && value instanceof OffsetTime)
+                || (prop instanceof LocalTime && value instanceof LocalTime)
+                || (prop instanceof DurationValue && value instanceof DurationValue)
+                || (prop instanceof String && value instanceof String)
+                || (prop instanceof Boolean && value instanceof Boolean)
+                || (prop instanceof Number && value instanceof Number)
+                || ((prop instanceof Collection || prop.getClass().isArray())
+                        && (value instanceof Collection || value.getClass().isArray()))
+                || (prop instanceof PointValue && value instanceof PointValue);
+    }
+
+    private boolean compareValues(Object prop, Object value) {
```

Comment:
Could use `org.neo4j.values.storable.Values.COMPARATOR` here to not have to maintain this in apoc?

```java
return switch (Values.COMPARATOR.ternaryCompare(Value.of(a), Value.of(b))) {
  case UNDEFINED -> ...;
  case EQUAL -> ...;
  case GREATER_THAN -> ...;
  case SMALLER_THAN -> ...;
}
```

Reply:
I did not know this existed, looks interesting thanks! I'll play with it :D